File: cpp/Leetcode/c++/problem41.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        map<int,int>m;
        for(int i:nums){
            m[i]=1;
        }
        for(int i=1;i<=nums.size();i++){
            if(m.count(i)==0){
                return i;
            }
        }
        return nums.size()+1;
    }
};

File: cpp/Leetcode/Python/problem41.py
class Solution:
    def firstMissingPositive(self, nums) -> int:
       d = {}
       for i in nums:
           d[i] = d.get(i, 0) + 1
       for i in range(1, len(nums) + 2):  # Check for missing positive integers
            if d.get(i, 0) == 0:
                return i
       return len(nums)+1